src/Service/EnterpriseSearchService.php

```diff
@@ -471,7 +478,22 @@ private function fetchEngines(): array
             throw new IndexingServiceException('Invalid response format for listEngines; missing "results"');
         }
 
-        $results = $response['results'] ?? [];
+        return $response;
+    }
+
+    private function fetchEngines(): array
+    {
+        $response = $this->fetchPaginatedEngines(1);
+
+        $results = $response['results'];
+
+        if (isset($response['meta']['page']['total_pages']) && $response['meta']['page']['total_pages'] > 1) {
+            foreach (range(2, $response['meta']['page']['total_pages']) as $page) {
+                $paginatedResponse = $this->fetchPaginatedEngines($page);
+
+                $results = array_merge($results, $paginatedResponse['results']);
```

Comment:
```suggestion
                $results = array_merge($results, $paginatedResponse['results'] ?? []);
```

Reply:
The function errors if this doesn't exist so this would be redundant wouldn't it?

```
if (!array_key_exists('results', $response)) {
    throw new IndexingServiceException('Invalid response format for listEngines; missing "results"');
}
```